src/cpp/session/modules/SessionPath.cpp

```diff
@@ -66,19 +66,19 @@ void safeReadPathsFromFile(const FilePath& filePath,
       LOG_ERROR(error);
 }
 
-void addToPathIfNecessary(const std::string& entry, std::string* pPath)
+void addToPathIfNecessary(
+      const std::string& entry,
+      std::vector<std::string>* pPathComponents)
 {
-   // guard against user input which may not parse as a valid regular expression
-   try
-   {
-      if (!regex_search(*pPath, boost::regex("(^|:)" + entry + "/?($|:)")))
-      {
-         if (!pPath->empty())
-            pPath->push_back(':');
-         pPath->append(entry);
-      }
-   }
-   CATCH_UNEXPECTED_EXCEPTION;
+   auto it = std::find(
+            pPathComponents->begin(),
+            pPathComponents->end(),
+            entry);
```

Comment:
It looks like we used to tolerate a trailing slash in the entry (i.e. `foo` matched `foo/`) but don't any more -- intentional?

Reply:
That wasn't intentional -- I'll fix that up.